modules/mono/mono_gd/gd_mono.cpp

```diff
@@ -501,26 +501,51 @@ bool GDMono::_load_project_assembly() {
 #endif
 
 #ifdef GD_MONO_HOT_RELOAD
+void GDMono::reload_failure() {
+	if (++project_load_failure_count >= 5) {
```

Comment:
Why are we attempting to reload 5 times? If it fails once why would it work the next time?

Reply:
Tbh it has been too long I don't remember exactly why, but IIRC the design of assembly reloading means that currently, if the editor integration wouldn't break it would retry forever and coupled with the first version of this PR loosing the reference to the old assembly and appearing to work on the second attempt, I just left the retry behavior as-is and put a cap on it.

It could theoretically work after a few attempts if the current unloading was still doing something that just took more time and wasn't blocked completely, but in most cases unloading will succeed in maybe 10 GCs or it will never work.

Reply:
> It could theoretically work after a few attempts if the current unloading was still doing something that just took more time and wasn't blocked completely

That's what I had in mind, but I'm not sure how likely this scenario is.

>  I just left the retry behavior as-is and put a cap on it.

That makes sense, I was only suggesting to go even further and make the cap 1 instead of 5. But I guess it's safer to make multiple attempts just in case.

Reply:
I made this configurable as a project setting and reduced the default to 3 reload attempts.